/**
 * Copyright (c) 2019 Greg Padiasek
 * Distributed under the terms of the MIT License.
 * See accompanying file LICENSE or copy at http://opensource.org/licenses/MIT
 */

//
// EAGLE messages
//

syntax = "proto2";
package eagle;

// GetFitness request

message GetFitness {
}

// GetFitness response

message GetFitnessResponse {
  required float fitness = 1;
}

// GetGraph request

message GetGraph {
}

// GetGraph response

message GetGraphResponse {
  required float fitness = 1;
  required bytes graph = 2;
}

// SetGraph request

message SetGraph {
  required float fitness = 1;
  required bytes graph = 2;
}


// Success response

message SuccessResponse {
}

// Failure response

message ErrorResponse {
  required uint32 status = 1;  
  required string message = 2;
}

// Message envelopes (union)

message Request {
  oneof request {
    eagle.GetFitness get_fitness = 10;
    eagle.GetGraph get_graph = 11;
    eagle.SetGraph set_graph = 12;
  }
}

message Response {
  oneof response {
    eagle.GetFitnessResponse get_fitness = 10;
    eagle.GetGraphResponse get_graph = 11;
    eagle.SuccessResponse success = 12;
    eagle.ErrorResponse error = 13;
  }
}

